### PR TITLE
Adds beginRefund functions

### PR DESCRIPTION
**Pull request by vegaro**
Added:
- `Purchases.beginRefundRequestForActiveEntitlement`
-  `Purchases.beginRefundRequestForEntitlement`
-  `Purchases.beginRefundRequestForProduct`

### CODE REVIEW

File: apiTester/purchases.ts [hunk 1]
```diff
@@ -162,3 +164,12 @@ function checkSyncObserverModeAmazonPurchase(productID: string,
...
     productID, receiptID, amazonUserID, isoCurrencyCode, price);
 }
+
+function checkBeginRefundRequest(
+  entitlementInfo: PurchasesEntitlementInfo,
+  purchasesStoreProduct: PurchasesStoreProduct
+) {
+  Purchases.beginRefundRequestForActiveEntitlement((refundRequestStatus: REFUND_REQUEST_STATUS) => {}, errorCallback);
+  Purchases.beginRefundRequestForEntitlement(entitlementInfo, (refundRequestStatus: REFUND_REQUEST_STATUS) => {}, errorCallback);
+  Purchases.beginRefundRequestForProduct(purchasesStoreProduct, (refundRequestStatus: REFUND_REQUEST_STATUS) => {}, errorCallback);
```
Comment (tonidero) [inline apiTester/purchases.ts:174]: Nitpick. I guess we don't have line length linters right? This seems pretty long.

File: examples/cordova-sample/MyApp/www/js/index.js [hunk 2]
```diff
@@ -242,32 +245,92 @@ const app = {
...
+      setStatusLabelText(error);
+    });
+  },
+
+  beginRefundRequestForEntitlement: function() {
+    setStatusLabelText("beginning refund request by entitlement id");
+    Purchases.getCustomerInfo(
+      customerInfo => {
+        if (!customerInfo.entitlements.active) {
+          setStatusLabelText("customer info doesn't have an active entitlements");
```
Comment (tonidero) [inline examples/cordova-sample/MyApp/www/js/index.js:288]:
```suggestion
          setStatusLabelText("customer info doesn't have active entitlements");
```

File: examples/cordova-sample/MyApp/www/js/index.js [hunk 3]
```diff
@@ -242,32 +245,92 @@ const app = {
...
+      }
+    );
+  },
+
+  beginRefundRequestForProduct: function() {
+    setStatusLabelText("beginning refund request for product");
+    Purchases.getCustomerInfo(
+      customerInfo => {
+        if (!customerInfo.entitlements.active) {
+          setStatusLabelText("customer info doesn't have an active entitlements");
```
Comment (tonidero) [inline examples/cordova-sample/MyApp/www/js/index.js:311]:
```suggestion
          setStatusLabelText("customer info doesn't have active entitlements");
```

File: examples/cordova-sample/MyApp/www/js/index.js [hunk 4]
```diff
@@ -242,32 +245,92 @@ const app = {
...
+        });
+      },
+      error => {
+        setStatusLabelText(error);
+      }
+    );
+  },
+
+  beginRefundRequestForProduct: function() {
+    setStatusLabelText("beginning refund request for product");
```
Comment (tonidero) [inline examples/cordova-sample/MyApp/www/js/index.js:307]: Hmm we should probably mimic what we do for the entitlement version. Maybe 
```suggestion
    setStatusLabelText("beginning refund request by product id");
```
Same below.

File: src/android/PurchasesPlugin.java [hunk 5]
```diff
@@ -416,6 +416,24 @@ private void presentCodeRedemptionSheet(CallbackContext callbackContext) {
...
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "beginRefundRequestForEntitlementId")
+    private void beginRefundRequestForEntitlementId(CallbackContext callbackContext) {
+        // noop
+        callbackContext.success();
+    }
+
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "beginRefundRequestForProductId")
+    private void beginRefundRequestForProductId(CallbackContext callbackContext) {
+        // noop
+        callbackContext.success();
```
Comment (tonidero) [inline src/android/PurchasesPlugin.java:434]: Hmm should we error out if calling these in android?

File: src/ios/PurchasesPlugin+Purchasing.swift [hunk 6]
```diff
@@ -118,10 +118,105 @@ import PurchasesHybridCommon
...
+//        return ^(RCErrorContainer * _Nullable error) {
+//            if (error == nil) {
+//                resolve(0);
+//            } else if ([error.info[@"userCancelled"] isEqual:@YES]) {
+//                resolve(1);
+//            } else {
+//                [self rejectPromiseWithBlock:reject error:error];
+//            }
+//        };
+//    }
```
Comment (tonidero) [inline src/ios/PurchasesPlugin+Purchasing.swift:219]: Need to remove these?

File: src/plugin/plugin.ts [hunk 7]
```diff
@@ -1487,6 +1511,99 @@ class Purchases {
...
   }
 
+  /**
+   * iOS 15+ only. Presents a refund request sheet in the current window scene for
+   * the latest transaction associated with the active entitlement.
+   *
+   * If the request was unsuccessful, no active entitlements could be found for
+   * the user, or multiple active entitlements were found for the user,
+   * the promise will return an error.
+   * If called in an unsupported platform (Android or iOS < 15), an `UnsupportedPlatformException` will be thrown.
```
Comment (tonidero) [inline src/plugin/plugin.ts:1521]: I think right now this will success if called on Android? Also, this name might not apply on iOS.

**Review by tonidero — COMMENTED** [review]
Mostly some nitpicks but I think we should fix the docs

File: src/android/PurchasesPlugin.java (hunk 5 above)
Comment (vegaro) [reply src/android/PurchasesPlugin.java:434]: we don't error out in any other method that's only available in iOS, so I don't think we should have some functions that error and some that don't. I think we should update all of them at once and do error out though, but that's a breaking change so we should do that in the next major probably. Do you agree? If you strongly feel we should error here, I will add it

File: src/plugin/plugin.ts (hunk 7 above)
Comment (vegaro) [reply src/plugin/plugin.ts:1521]: right, I copied this from React Native, I will update

**Comment by vegaro** [comment]
Fixed those comments. Sorry for not re-reviewing before asking for review

**Review by tonidero — APPROVED** [review]

**Pull request merged**